### ISSUE TITLE
Cuda compiler fails w/ threads dep

### ISSUE THREAD

**Issue by jml1795** [mesonbuild/meson#4999]
Looks like cuda fails when linking against a library w/ a threads dep.

Can't submit PR now but patch is below. Will submit PR tonight unless somebody wants to push this
```sh
diff --git a/mesonbuild/compilers/cuda.py b/mesonbuild/compilers/cuda.py
index d1964fd..25453fc 100644
--- a/mesonbuild/compilers/cuda.py
+++ b/mesonbuild/compilers/cuda.py
@@ -46,6 +46,12 @@ class CudaCompiler(Compiler):
     def get_no_stdinc_args(self):
         return []

+    def thread_link_flags(self, environment):
+        return []
+
```

**Comment by obilaniu** [comment]
Would it not be better to use `-Xcompiler -pthread` or similar?

**Comment by jml1795** [comment]
@obilaniu yes, thanks. Are cuda tests ignored in CI?

**Comment by obilaniu** [comment]
@jml1795 They're being skipped because there's no GPU-enabled CI machines, and none of the test-cases would have discovered this anyways because none use threads.

**Comment by jml1795** [comment]
@obilaniu I added a test case for threads, just wasn't sure about whether it would actually get run

**Comment by jml1795** [comment]
@obilaniu Closing this issue as #5000 was merged. I'm assuming I should do this since I was the OP?

**Comment by obilaniu** [comment]
@jml1795 If, within your commit message, you add text such as `Closes #4999` (Note the `#` and the use of the word `closes`; I think `fixes` may work as well), then when it is merged, the Github issue so referenced is automatically closed.
